R/durations.r

```diff
@@ -242,28 +245,25 @@ setMethod("[[<-", signature(x = "Duration"),
 #' boundary + ddays(1) # duration
 #' @export
 duration <- function(num = NULL, units = "seconds", ...) {
-  nums <- list(...)
   if (is.character(num)) {
     as.duration(parse_period(num))
-  } else if (!is.null(num) && length(nums) > 0) {
-    c(.duration_from_num(num, units), .duration_from_units(nums))
-  } else if (!is.null(num)) {
-    .duration_from_num(num, units)
-  } else if (length(nums)) {
-    .duration_from_units(nums)
   } else {
-    stop("No valid values have been passed to 'duration' constructor")
+    c(.duration_from_num(num, units),
+      .duration_from_units(list(...)))
   }
 }
 
 .duration_from_num <- function(num, units) {
+  if (length(num) == 0)
+    return(new("Duration", numeric()))
+
   if (!is.numeric(num)) {
     stop(sprintf("First argument to `duration` constructor must be character or numeric. Supplied object of class '%s'", class(num)))
   }
 
   ## qucik check for common wrongdoings: https://github.com/hadley/lubridate/issues/462
   if (class(num)[[1]] %in% c("Interval", "Period"))
-    stop("Interval or Period objects cannot be used as input to 'period' constructor. Plese use 'as.duration'.")
+    stop("Interval or Period objects cannot be used as input to 'duration' constructor. Use 'as.duration' instead.")
```

Comment:
If you're touching this, I think you should update to:

```suggestion
    stop("Interval or Period objects cannot be used as input to `duration()` constructor. Use `as.duration()` instead.", call. = FALSE)
```